lib/puppet/configurer.rb

```diff
@@ -161,6 +161,7 @@ def run_internal(options)
         begin
           if node = Puppet::Node.indirection.find(Puppet[:node_name_value],
               :environment => Puppet::Node::Environment.remote(@environment),
+              :configured_environment => Puppet[:environment],
```

Comment:
Ah, yep, I forgot the compiler terminus also calls `Puppet::Node.indirection.find`, and in that case, the environments can be different. :+1: for being explicit.